### PR TITLE
ScreenSaver: Delay footer/header repaint if screensaver_delay is enabled

### LINKED ISSUE #7327: Status bar does not wait for screensaver delay
* KOReader version: v2021.02
* Device: Kobo Aura

#### Issue

With a delay to disable the screensaver image, the status bar appears immediately after waking up the device (i.e. opening the sleep cover), overwriting part of the screensaver image. Then after the set delay the screensaver image disappears (this part is OK).

#### Steps to reproduce

* Enable normal status bar (not tested with "alternate")
* Enable screensaver and "delay when exit from screensaver" (this doesn't sound gramatically correct, by the way)
* Activate the screensaver by suspending (putting to sleep) the device
* Wake up the device
* Look how the status bar "corrupts" the lower part of your nice screensaver image :D

### PR DESCRIPTION
**Pull request by NiLuJe**
Fix #7327

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/7334)

### CODE REVIEW

File: frontend/ui/elements/screensaver_menu.lua [hunk 1]
```diff
@@ -208,7 +208,7 @@ return {
                 }
             },
             {
-                text = _("Delay when exit from screensaver"),
+                text = _("Let the screensaver linger on wakeup"),
```
Comment (Frenzie) [inline frontend/ui/elements/screensaver_menu.lua:211]: rofl

Definitely amusing, but maybe just `leave screensaver after waking up` or something?

**Review by Frenzie — APPROVED** [review]

File: frontend/apps/reader/modules/readerfooter.lua [hunk 2]
```diff
@@ -2177,6 +2177,25 @@ function ReaderFooter:refreshFooter(refresh, signal)
...
+    self:rescheduleFooterAutoRefreshIfNeeded()
+end
+
+function ReaderFooter:onOutOfScreenSaver()
+    if not self._delayed_screensaver then
+        return
+    end
+
+    self._delayed_screensaver = nil
+
```
Comment (poire-z) [inline frontend/apps/reader/modules/readerfooter.lua:2198]: Huge file, more blank lines, more huge file :) (this line does not need to stand out like that)

**Review by poire-z — APPROVED** [review]
(Dunno what this screensaver option use case is... Resuming when the lights are out, and wanting to see the cover for a few seconds?)

**Comment by Jellby** [comment]
> (Dunno what this screensaver option use case is... Resuming when the lights are out, and wanting to see the cover for a few seconds?)

In my case, I have nice pictures as screensaver, but I never see them properly, because with a sleep cover they're only shown while the device is covered. With the delay option, I get to enjoy the pictures for a few seconds, it's not like I'm in a rush to start reading anyway.

If I didn't have a sleep cover and had to press some button to wake up, I'd disable the delay indeed.

**Comment by NiLuJe** [comment]
Are we all happy with the reworded options?

File: frontend/ui/elements/screensaver_menu.lua [hunk 3]
```diff
@@ -208,7 +208,7 @@ return {
                 }
             },
             {
-                text = _("Delay when exit from screensaver"),
+                text = _("Keep displaying the screensaver after wakeup"),
```
Comment (poire-z) [inline frontend/ui/elements/screensaver_menu.lua:211]: `Keep screensaver displayed after wakeup` ?
(Somehow, this -ing feels strange for our menus, dunno if we have any :)
Fine with the others.

**Review by poire-z — APPROVED** [review]

**Pull request merged**